src/forms/Slushbucket.tsx

```diff
@@ -0,0 +1,237 @@
+import React, { Component } from "react";
+
+import MDSpinner from "react-md-spinner";
+import Scrollbars from "react-custom-scrollbars";
+import ClearButton from "../components/ClearButton.js";
+
+import styles from "./SlushBucket.css";
+import formStyles from "../styles/Forms.css";
+import inputStyles from "../styles/Input.css";
+import displayStyles from "../styles/Display.css";
+
+type choiceT = {value: string | number, display: string};
+export type choicesT = [choiceT];
+type valueItemT = string | number
+type valueT = [valueItemT];
+
+interface Props {
+  title: string,
+  name: string,
+  placeholder: string,
+  readonly: boolean | undefined,
+  isFetching: boolean | undefined,
+  value: valueT,
+  choices: choicesT,
+  validators?: Function[],
+  validated: boolean,
+  valueChanged: Function,
+};
+
+interface State {
+  searchString: string
+}
+
+export default class SlushBucket extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      searchString: ""
+    };
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+  }
+
+  handleKeyUp(e: any): void {
+    if (e.key === "Escape") this.resetSearchString();
+  }
+
+  resetSearchString() {
+    this.setState({ searchString: "" });
+  }
+
+  handleInput(e: any): void {
+    this.setState({ searchString: e.target.value });
+  }
+
+  render() {
+    const {
+      choices,
+      value,
+      isFetching,
+      placeholder,
```

Comment:
I called this props placeHolder everywhere else, but now I notice that that's actually incorrect (it's one word in English). So needs to be replaced in the other files.

Here it's a bit unusual as this is the placeholder for the search field, not the actual input field. I think I'd prefer to have a hourglass icon there and just the word "Search", but that's not a strong opinion.